polkadot/consensus/src/lib.rs

```diff
@@ -65,6 +67,9 @@ pub use self::error::{ErrorKind, Error};
 
 mod error;
 
+// block size limit.
+const MAX_TRANSACTIONS_SIZE: usize = 4 * 1024 * 1024;
```

Comment:
Shouldn't the block size limit be defined by the runtime?

Reply:
we can move it in there eventually, but I don't see much difference between that and having it negotiated off-chain somehow.

Reply:
yeah - should definitely be a chain param defined by the runtime. i guess that's one more thing to go into the runtime/native API.